draft-ietf-jsonpath-base.md

```diff
@@ -1535,36 +1523,80 @@ A function expression is well-typed if all of the following are true:
    * The argument is a literal primitive value and the defined type of the parameter is `ValueType`.
    * The argument is a Singular Path or `filter-path` (which includes
      Singular Paths), or a function expression with declared result
-     type `NodesType`.  Where the declared type of the parameter is
-     not `NodesType`, a conversion applies.
+     type `NodesType`.
+
+{{tab-exist-value}} summarizes how the two conversion functions can be
+used to make use of functions returning nodelists, using a
+hypothetical function `fn` of declared type `NodesType`: The table
+gives the result of the conversion function, and the effect of that
+result on the expression given. (`fn` is a hypothetical function
+extension that takes and results in `NodesType`; this function will
+also be used in example further below.)
 
-#### Conversion example
-{:unnumbered}
 
-While functions returning `NodesType` can appear in both comparisons and test expressions,
-path authors must be aware of the conversions and the potential outcomes.
-Specifically, nodelists convert to values and logicals according to {{tbl-typeconv}}:
+| fn(...) result         | `exist(fn(@..a))` evaluation (as a test expression) | `value(fn(@..a))==42` evaluation (in a comparison)             |
+| :-:                    | :-                                                  | :-                                                             |
+| empty nodelist         | `LogicalFalse`.<br>The test fails.                     | `Nothing`.<br>The expression result is false.                     |
+| single-node nodelist   | `LogicalTrue`.<br>The test succeeds.                   | The node's value.<br>The expression result is true if that is 42. |
+| multiple-node nodelist | `LogicalTrue`.<br>The test succeeds.                   | `Nothing`.<br>The expression result is false.                     |
+{: #tab-exist-value title="Using a function returning a `NodesType` in
+test expressions and comparisons"}
 
-| Function return        | Converted as Value    | Converted as Logical |
-| :-:                    | :-:                   | :-:                  |
-| empty nodelist         | `Nothing`             | `LogicalFalse`       |
-| single-node nodelist   | the node's JSON value | `LogicalTrue`        |
-| multiple-node nodelist | `Nothing`             | `LogicalTrue`        |
-{: #tbl-typeconv title="Conversion of nodelists to values and logicals"}
+Note that a multiple-node nodelist result is deemed `LogicalTrue` in
+the test expression, but cannot be used for obtaining a single
+value and thus feeds the comparison expression with `Nothing`.
```

Comment:
This is confusing. Shouldn't this text be part of the spec of the `value()` function instead?

Reply:
Went from "the" to "a".